src/pkg/util/env/clean.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package env
+
+import (
+	"strings"
+
+	"github.com/opencontainers/runtime-tools/generate"
+	"github.com/singularityware/singularity/src/pkg/sylog"
+)
+
+// CleanEnv cleans environment variables before running the container
+func CleanEnv(g *generate.Generator, IsCleanEnv bool, environment []string) {
+
+	for _, env := range environment {
+		e := strings.SplitN(env, "=", 2)
+		if len(e) != 2 {
+			sylog.Verbosef("Can't process environment variable %s", env)
+			continue
+		}
+
+		// Transpose environment
+		if strings.HasPrefix(e[0], "SINGULARITYENV_") {
```

Comment:
Also, since this is in an if...else if block, this implies that `SINGULARITYENV_` variables are *always* transformed even when `--cleanenv` is set, is that what we want?

Reply:
Yes that is the behavior in 2.6.